src/fides/api/ops/task/graph_task.py

```diff
@@ -765,34 +774,45 @@ def termination_fn(*dependent_values: int) -> Tuple[int, ...]:
                 access_request_data.get(
                     str(k), []
                 ),  # Pass in the results of the access request for this collection
-                *[
+                [
                     access_request_data.get(
                         str(upstream_key), []
                     )  # Additionally pass in the original input data we used for the access request. It's helpful in
                     # cases like the EmailConnector where the access request doesn't actually retrieve data.
                     for upstream_key in t.input_keys
                 ],
+                *_evaluate_erasure_dependencies(t, end_nodes),
```

Comment:
Major change: adding the dependent tasks to each task in the task graph

Reply:
The main thing I'd watch out for here is that we've done some check to make sure this is resolvable. I'd add a big picture check that ensures that we're not saying "erase A before B", and elsewhere saying "erase B before A".

Reply:
Using `getcycle` from `dask.core` to detect cycles and raise a `TraversalError`

```
# using dask.core function to detect cycles in the generated graph
collection_cycle = getcycle(dsk, None)
if collection_cycle:
    raise TraversalError(
        f"The values for the `erase_after` fields caused a cycle in the following collections {collection_cycle}"
    )
```

Reply:
Nice use of dask's existing method here